core/DataTable.php

```diff
@@ -1573,7 +1576,12 @@ public static function makeFromIndexedArray($array, $subtablePerLabel = null)
             if (isset($subtablePerLabel[$label])) {
                 $cleanRow[Row::DATATABLE_ASSOCIATED] = $subtablePerLabel[$label];
             }
-            $table->addRow(new Row($cleanRow));
+
+            if ($arrayHasSummaryRow && $label == self::LABEL_SUMMARY_ROW) {
```

Comment:
Do we need `$arrayHasSummaryRow ` because `LABEL_SUMMARY_ROW` could be an actual label within the table?

Reply:
Yes.

Reply:
Not sure I'm getting it fully. If there's a label `-1` then it would still aggregate it maybe wrongly? It be only helping if there are reports that never have a summary row but potentially a `-1` value? Which might be rarely the case? So maybe we could just execute it always? Not sure I'm fully understanding it

Reply:
So DataTable identifies the summary row by ID, not by label (the ID is also -1). I assume this is because the label could be -1 and still be a valid dimension value (eg, for custom variable value/custom dimension). So we can't do this in every case, only when using ranking query and we know it's the right label. Alternatively, we could change the others label in RankingQuery to something like `__ranking_query_others__`, and always check for that.

Reply:
Not quite sure I understand as you mention the row by ID. Or maybe I wasn't clear what I meant. I reckon we could change 

`if ($arrayHasSummaryRow && $label == self::LABEL_SUMMARY_ROW)`
to 
`if ($label == self::LABEL_SUMMARY_ROW)`

and it wouldn't really make a difference?

Reply:
It would make a difference because the label could legitimately be `-1` in certain cases, like custom variable value / custom dimension value. Not sure about other reports, but this would essentially be saying that `-1` could never be used as a dimension value for any dimension, because it is used to denote a summary row.

Reply:
Wouldn't CustomVariables or CustomDimensions also set `$arrayHasSummaryRow = true` because we also want to group these? Technically it could have a summary row as well?

Reply:
Any report that uses RankingQuery would use `$arrayHasSummaryRow`, however, it would be a problem for reports where `-1` could be a valid label in the report.

Reply:
OK 👍  thought custom variables and custom dimensions would maybe also already use rankingQuery but they don't. Eventually all archivers should probably use the ranking query though (not going to happen any time soon though so can be ignored probably).

Custom Reports uses ranking query and possibly `-1` could be a valid label but I suppose we can't avoid this unless as you say we set it to something like `__ranking_query_others__ ` at some point. But even then I'm not sure it can be actually avoided so much, unless we also archived it using that label.

Reply:
This code is executed before archiving so a special, very unlikely label like `__ranking_query_others__` should work (the summary row w/ ID = -1 is what is archived). I can look into that.